src/nvim/option.c

```diff
@@ -3485,36 +3528,254 @@ vimoption_T *get_option(int opt_idx)
+    set_option_varp(opt_idx, varp, old_value, true);
+    // When resetting some values, need to act on it.
+    if (restore_chartab) {
+      (void)buf_init_chartab(curbuf, true);
+    }
+
+    // Unset new_value as it is no longer valid.
+    new_value = NIL_OPTVAL;  // NOLINT(clang-analyzer-deadcode.DeadStores)
+  } else {
+    // Re-assign the new value as its value may get freed or modified by the option callback.
+    new_value = optval_from_varp(opt_idx, varp);
+
+    // Remember where the option was set.
+    set_option_sctx_idx(opt_idx, opt_flags, current_sctx);
+    // Free options that are in allocated memory.
+    // Use "free_oldval", because recursiveness may change the flags (esp. init_highlight()).
+    if (free_oldval) {
+      optval_free(old_value);
+    }
+    opt->flags |= P_ALLOCED;
+
+    // Check the bound for num options.
+    if (new_value.type == kOptValTypeNumber) {
+      errmsg
+        = check_num_option_bounds((OptInt *)varp, old_value.data.number, errbuf, errbuflen, errmsg);
+      // Re-assign new_value because the new value was modified by the bound check.
+      new_value = optval_from_varp(opt_idx, varp);
+    }
+
+    if ((opt_flags & (OPT_LOCAL | OPT_GLOBAL)) == 0 && (opt->indir & PV_BOTH)) {
+      // Global option with local value set to use global value.
+      // Free the local value and clear it.
+      void *varp_local = get_varp_scope(opt, OPT_LOCAL);
+      OptVal local_unset_value = optval_unset_local(opt_idx, varp_local);
+      set_option_varp(opt_idx, varp_local, local_unset_value, true);
+    } else if ((opt_flags & (OPT_LOCAL | OPT_GLOBAL)) == 0) {
+      // May set global value for local option.
+      void *varp_global = get_varp_scope(opt, OPT_GLOBAL);
+      set_option_varp(opt_idx, varp_global, optval_copy(new_value), true);
+    }
+  }
+
+  // Skip processing the option further if asked to do so.
+  if (*doskip) {
+    return errmsg;
+  }
+
+  if (errmsg == NULL) {
+    // Trigger the autocommand only after setting the flags.
+    if (varp == &curbuf->b_p_syn) {
+      do_syntax_autocmd(curbuf, value_changed);
+    } else if (varp == &curbuf->b_p_ft) {
+      // 'filetype' is set, trigger the FileType autocommand
+      // Skip this when called from a modeline
+      // Force autocmd when the filetype was changed
+      if (!(opt_flags & OPT_MODELINE) || value_changed) {
+        do_filetype_autocmd(curbuf, value_changed);
+      }
+    } else if (varp == &curwin->w_s->b_p_spl) {
+      do_spelllang_source(curwin);
+    }
+  }
+
+  // In case 'columns' or 'ls' changed.
+  comp_col();
+
+  if (varp == &p_mouse) {
+    setmouse();  // in case 'mouse' changed
+  } else if ((varp == &p_flp || varp == &(curbuf->b_p_flp)) && curwin->w_briopt_list) {
+    // Changing Formatlistpattern when briopt includes the list setting:
+    // redraw
+    redraw_all_later(UPD_NOT_VALID);
+  } else if (varp == &p_wbr || varp == &(curwin->w_p_wbr)) {
+    // add / remove curwindow bars for 'curwinbar'
```

Comment:
`add / remove curwin bars for 'winbar'` ?

Reply:
I think it's intended to be "window bars"